Update assets documentation (#416)
osisoft/OCS-Docs: Content_Portal/Documentation/Assets/assets-api.md

If the asset you are trying to create references an asset type (through the AssetTypeId property) and if the corresponding asset type has a metadata value with the same `Id`, then the name and SDS type code of the metadata value on the asset must be null. If the asset type does not have metadata value with a corresponding `Id`, the name and SDS type code on the asset cannot be null.

> To support flexibility, on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is from an asset type.

Reviewer: ```suggestion
To support flexibility on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is created from an asset type.
```

> 

Reviewer: ### Examples

> | Example                                             | Metadata value on asset type                          |                                              |

Reviewer: ```suggestion
| Asset values                                             | Metadata values on the asset type                          |                                              |
```

Author: That change looks fine to me.

| ----------------------------------------------------| ----------------------------------------------------- | ------------------------------------------------- | 

> | Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |

Reviewer: ```suggestion
| Id = "Metadata Id"​ and Name = Not specified         | Id matches asset type metadata Id | Uses matching asset type metadata name |
```

> | Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |

Reviewer: ```suggestion
| `Id` = ***Metadata Id***​ and Name = Not specified         | `Id` matches the asset type's Metadata Id | Uses matching asset type Metadata Name |
```

> | Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |

Reviewer: ```suggestion
| Id = "Metadata Id"​ and Name = Not specified         | Id matches the asset type's Metadata Id | Uses the matching asset type's Metadata Name |
```

> |                                                     | Id does not match any Asset Type Metadata Id | Id is used as Name |

Reviewer: ```suggestion
|                                                     | Id does not match any asset type Metadata Id | Id is used as Name |
```

> | Id = Not specified​ and Name = "Metadata Name"       | Name matches Asset Type Metadata Name | Use matching Asset Type Metadata Id |

Reviewer: ```suggestion
| Id = Not specified​ and Name = "Metadata Name"       | Name matches asset type metadata name | Use matching asset type metadata Id |
```
Anthony,  Why do you have "Metadata Name" in quotes? Is this to indicate that this is a parameter? Is Name a parameter?

Author: I think perhaps that was an initial mistake. Thanks for updating.

> |                                                     | Name does not match any Asset Type Metadata Name | GUID is generated for Id |

Reviewer: ```suggestion
|                                                     | Name does not match any asset type Metadata Name | GUID is generated for Id |
```

> | Id = "Metadata Id"​ and Name = "Metadata Name"       | Only Id or Name (not both) matches Asset Type Metadata Id or Name | Error. |

Reviewer: ```suggestion
| Id = "Metadata Id"​ and Name = "Metadata Name"       | Either Id or Name, but not both, match asset type Metadata Id or Name | Error. |
```

Create or update an asset with a specified `Id`. If the asset already exists, you can specify an If-Match property in the HTTP request header to ensure that the asset is modified only if its version matches.

To support flexibility, on creation and update, the following rules and behaviors are executed for metadata and stream references on a given asset when that asset is from an asset type.

> 

Reviewer: ###Examples

> | Example                                             | Metadata value on asset type                          |                                              |

Reviewer: ```suggestion
| Asset values                                            | Metadata values on the asset type                          |                                              |
```

| ----------------------------------------------------| ----------------------------------------------------- | ------------------------------------------------- | 

> | Id = "Metadata Id"​ and Name = Not specified         | Id matches Asset Type Metadata Id | Uses matching Asset Type Metadata Name |

Reviewer: ```suggestion
| Id = "Metadata Id"​ and Name = Not specified         | Id matches the asset type's Metadata Id | Uses the matching asset type Metadata Name |
```

> |                                                     | Id matches existing Asset Type Metadata Id | Uses existing matching Metadata Name|

Reviewer: ```suggestion
|                                                     | Id matches the existing asset type's Metadata Id | Uses the existing matching Metadata Name|
```

> |                                                     | Id does not match any Asset Type Metadata Id | Id is used as Name |

Reviewer: ```suggestion
|                                                     | Id does not match any asset type Metadata Id | Id is used as Name |
```

> | Id = Not specified​ and Name = "Metadata Name"       | Name matches Asset Type Metadata Name | Use matching Asset Type Metadata Id |

Reviewer: ```suggestion
| Id = Not specified​ and Name = "Metadata Name"       | Name matches the asset type's Metadata Name | Uses the matching asset type Metadata Id |
```

Author: Looks good. Thanks

> |                                                     | Name matches existing Asset Type Metadata Name | Use matching existing Asset Type Metadata Id |

Reviewer: ```suggestion
|                                                     | Name matches the existing asset type's Metadata Name | Uses the matching existing asset type Metadata Id |
```

Author: Looks good. Thanks

> |                                                     | Name does not match any Asset Type Metadata Name | GUID is generated for Id |

Reviewer: ```suggestion
|                                                     | Name does not match any asset type Metadata Name | GUID is generated for Id |
```

> | Id = "Metadata Id"​ and Name = "Metadata Name"       | Only Id or Name (not both) matches Asset Type Metadata Id or Name | Error. |

Reviewer: ```suggestion
| Id = "Metadata Id"​ and Name = "Metadata Name"       | Only Id or Name (not both) matches asset type Metadata Id or Name | Error. |
```

> |                                                     | Only Id or Name (not both) matches existing Asset Type Metadata Id or Name | The non-matching Id or Name of the existing Metadata is updated |

Reviewer: ```suggestion
|                                                     | Only Id or Name (not both) matches existing asset type Metadata Id or Name | The non-matching Id or Name of the existing Metadata is updated |
```
